opc-ua-sdk/sdk-client/src/main/java/org/eclipse/milo/opcua/sdk/client/subscriptions/OpcUaSubscriptionManager.java

```diff
@@ -351,6 +351,79 @@ private UInteger getLifetimeCount(double publishingInterval, UInteger maxKeepAli
         });
     }
 
+    @Override
+    public CompletableFuture<UaSubscription> transferSubscription(
+        UaSubscription subscription,
+        ItemTransferredCallback callback
+    ) {
+
+        OpcUaSubscription newSubscription = new OpcUaSubscription(
+            client,
+            subscription.getSubscriptionId(),
+            subscription.getRevisedPublishingInterval(),
+            subscription.getRevisedLifetimeCount(),
+            subscription.getRevisedMaxKeepAliveCount(),
+            subscription.getMaxNotificationsPerPublish(),
+            subscription.isPublishingEnabled(),
+            subscription.getPriority()
+        );
+
+        newSubscription.setRequestedLifetimeCount(subscription.getRequestedLifetimeCount());
+        newSubscription.setRequestedMaxKeepAliveCount(subscription.getRequestedMaxKeepAliveCount());
+        newSubscription.setRequestedPublishingInterval(subscription.getRequestedPublishingInterval());
+
+        subscriptions.put(newSubscription.getSubscriptionId(), newSubscription);
+
+        return newSubscription
+            .transferMonitoredItems(subscription, callback)
+            .thenApply(v -> newSubscription);
+    }
+
+    @Override
+    public CompletableFuture<UaSubscription> transferSubscription(
+        UInteger subscriptionId,
+        UInteger nextSequenceNumber,
+        double requestedPublishingInterval,
+        double publishingInterval,
+        UInteger requestedLifetimeCount,
+        UInteger lifetimeCount,
+        UInteger requestedMaxKeepAliveCount,
+        UInteger maxKeepAliveCount,
+        UInteger maxNotificationsPerPublish,
+        boolean publishingEnabled,
+        UByte priority,
+        List<TransferMonitoredItemsRequest> itemsToTransfer
```

Comment:
I'm not convinced `itemsToTransfer` should be a List of `TransferMonitoredItemsRequest`. It already contains what should be all the monitored items for the subscription being transferred inside of it (`List<MonitoredItemTransferRequest>`).

Can you convince me otherwise @marcus-orchard ?

Reply:
@kevinherron - Sorry, was tied up on another project. Started looking at your comment this morning. Should be able to follow up shortly.

Reply:
OK, took me a few minutes but I remember my thinking in regards to this. 

I wanted to allow the ability to *easily* support different callbacks for groups of monitored items within a subscription should the original subscription have been created that way (or for some other unforeseen need).

``` 
    private final List<MonitoredItemTransferRequest> monitoredItems;
    private final ItemTransferredCallback itemTransferCallback;
```